plugins/GSdx/Renderers/SW/GSRasterizer.cpp

```diff
@@ -1086,8 +1087,10 @@ void GSRasterizer::Flush(const GSVertexSW* vertex, const uint32* index, const GS
 				int pixels = e->_pad.i32[0];
 				int left = e->_pad.i32[1];
 				int top = e->_pad.i32[2];
-
-				DrawScanline(pixels, left, top, *e++);
+				if (!m_scanline_mask || m_scanline_mask == 3 && (top & 1) == 1 || m_scanline_mask == 2 && (top & 1) == 0)
```

Comment:
Could probably use some brackets for clarity, I'm not sure how this would be interpreted in different scenarios